packages/grid/x-data-grid/src/components/GridPagination.tsx

```diff
@@ -31,38 +31,41 @@ export const GridPagination = React.forwardRef<HTMLDivElement, Partial<TablePagi
     const paginationState = useGridSelector(apiRef, gridPaginationSelector);
 
     const lastPage = React.useMemo(
-      () => Math.floor(paginationState.rowCount / (paginationState.pageSize || 1)),
-      [paginationState.rowCount, paginationState.pageSize],
+      () => Math.floor(paginationState.rowCount / (paginationState.paginationModel.pageSize || 1)),
+      [paginationState.rowCount, paginationState.paginationModel],
     );
 
     const handlePageSizeChange = React.useCallback(
       (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
-        const newPageSize = Number(event.target.value);
-        apiRef.current.setPageSize(newPageSize);
+        const pageSize = Number(event.target.value);
+        apiRef.current.setPaginationModel({ pageSize, page: paginationState.paginationModel.page });
```

Comment:
Would be nice to be able to do the following: specifying only the elements you want to update
```suggestion
        apiRef.current.setPaginationModel({ pageSize });
```

Same idea for `handlePageChange`

Reply:
You've created `setPage` and `setPageSize` for those purpose ?

Reply:
Yes, I kept them for easing up the migration pain a bit. It's opinionated but I was thinking to deprecate them and remove them in the next major release basically to keep the API simple.

Reply:
I don't have a strong opinion about keeping/removing those methods